jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -827,13 +827,15 @@ ecma_builtin_array_prototype_object_slice (ecma_value_t arg1, /**< start */
 
   JERRY_ASSERT (start <= len && end <= len);
 
+  bool use_fast_path = ecma_op_object_is_fast_array (obj_p);
```

Comment:
I'd move this to the `#else /* !ENABLED (JERRY_ES2015) */`

Reply:
Fast access arrays are both present in ES5 and ES6 profile. The empty array construction in ES5 always creates a fast array object, but in ES6 due to the species symbol there is no insurance that the returned array is fast mode array therefore the extra check is only needed here. But the `ecma_op_object_is_fast_array (obj_p);` is needed in both profiles.

Reply:
I see.